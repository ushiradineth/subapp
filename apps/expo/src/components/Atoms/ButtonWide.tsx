import React from "react";
import { Text } from "react-native";
import { Button } from "tamagui";

interface Props {
  icon?: React.ReactNode;
  text: string;
  onPress: () => void;
  justifyText?: "left" | "center" | "right";
}

const ButtonWide = ({ icon, text, onPress, justifyText = "left" }: Props) => {
  return (
    <Button onPress={onPress} className={`bg-background flex h-16 w-full items-center justify-${justifyText} rounded-3xl`}>
      {icon}
      <Text className="text-[16px] font-semibold">{text}</Text>
    </Button>
  );
};

export default ButtonWide;
