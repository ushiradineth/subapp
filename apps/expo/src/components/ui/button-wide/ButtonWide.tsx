import React from "react";
import { Text } from "react-native";
import { Button } from "tamagui";

interface Props {
  icon: React.ReactNode;
  text: string;
  onPress: () => void;
}

const ButtonWide = ({ icon, text, onPress }: Props) => {
  return (
    <Button onPress={onPress} className="bg-background flex h-16 w-full items-center justify-start rounded-3xl">
      {icon}
      <Text className="text-[16px] font-semibold">{text}</Text>
    </Button>
  );
};

export default ButtonWide;
