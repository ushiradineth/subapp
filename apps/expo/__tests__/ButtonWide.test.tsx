import React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import ButtonWide from "../src/components/Atoms/ButtonWide";
import config from "../tamagui.config";

describe("ButtonWide", () => {
  it("can handle onPress", () => {
    const handlePress = jest.fn();

    const { getByText } = render(
      <TamaguiProvider config={config}>
        <ButtonWide onPress={handlePress} text="Testing" />
      </TamaguiProvider>,
    );

    const button = getByText("Testing");

    fireEvent.press(button);
    expect(handlePress).toBeCalledTimes(1);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
