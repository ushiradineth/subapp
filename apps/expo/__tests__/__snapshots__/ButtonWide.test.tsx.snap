// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWide can handle onPress 1`] = `
<View
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(243,243,243,1.00)",
        "borderBottomColor": "rgba(0,0,0,0.00)",
        "borderBottomLeftRadius": 9,
        "borderBottomRightRadius": 9,
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(0,0,0,0.00)",
        "borderLeftWidth": 1,
        "borderRightColor": "rgba(0,0,0,0.00)",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "rgba(0,0,0,0.00)",
        "borderTopLeftRadius": 9,
        "borderTopRightRadius": 9,
        "borderTopWidth": 1,
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexShrink": 0,
        "flexWrap": "nowrap",
        "height": 44,
        "justifyContent": "center",
        "minHeight": 0,
        "minWidth": 0,
        "paddingLeft": 18,
        "paddingRight": 18,
        "position": "relative",
      },
    ]
  }
>
  <Text
    className="text-[16px] font-semibold"
  >
    Testing
  </Text>
</View>
`;
